python/lsst/pipe/analysis/plotRhoStatistics.py

```diff
@@ -0,0 +1,347 @@
+    if g1b is None or g2b is None:
+        # Calculate the auto-correlation
+        xy.process(catA)
+    else:
+        catB = treecorr.Catalog(ra=ra, dec=dec, g1=g1b, g2=g2b, ra_units="degrees",
+                               dec_units="degrees")
+        # Calculate the cross-correlation
+        xy.process(catA, catB)
+
+    return xy
+
+# RFC: Should we move this to utils.py?
+def calculateRhoStatistics(ra, dec, e1, e2, e1Res, e2Res, SizeRes, ra_units="degrees", dec_units="degrees",
+        **kwargs):
+
+        treecorr_kwargs = {'nbins':kwargs.get('nbins',10), 
+                        'min_sep':kwargs.get('min_sep',0.5), 
+                        'max_sep':kwargs.get('max_sep',10), 
+                        'sep_units':kwargs.get('sep_units',"arcmin"), 
+                        'verbose':kwargs.get('verbose',0)}
+
+        # Scale the SizeRes by ellipticities
+        e1SizeRes = e1*SizeRes
+        e2SizeRes = e2*SizeRes
+
+        # Package the arguments to capture auto-/cross-correlations for Rowe statistics
+        g_args = {1: (e1Res, e2Res, None, None),
+                  2: (e1, e2, e1Res, e2Res),
+                  3: (e1SizeRes, e2SizeRes, None, None),
+                  4: (e1Res, e2Res, e1SizeRes, e2SizeRes),
+                  5: (e1, e2, e1SizeRes, e2SizeRes) }
+
+        radii = dict.fromkeys(range(1,6))
+        xip = dict.fromkeys(range(1,6))
+        xipErr = dict.fromkeys(range(1,6))
+
+        # Pass the appropriate arguments to the correlator
+        for rhoIndex in range(1,6):
+            # RFC: Should I just implement the corr2 functionality here inline?
+            xy = corr2(ra, dec, *(g_args[rhoIndex]), ra_units=ra_units, dec_units=dec_units, **treecorr_kwargs)
+            radii[rhoIndex] = np.exp(xy.meanlogr)
+            xip[rhoIndex] = xy.xip
+            xipErr[rhoIndex] = np.sqrt(xy.varxi)
+
+        return radii, xip, xipErr
+        
+def plotEllipResCorrVisits(butler, visits, filter, filename):
+    fig, axes  = plt.subplots(1,2)
+    fig.set_size_inches(16,9)
+
+    shapeCol = "ext_shapeHSM_HsmSourceMoments"
+    psfShapeCol = "ext_shapeHSM_HsmPsfMoments"
+    e1Func = E1(shapeCol)
+    e2Func = E2(shapeCol)
+    e1ResidsFunc = E1Resids(shapeCol, psfShapeCol)
+    e2ResidsFunc = E2Resids(shapeCol, psfShapeCol)
+    TraceSizeResidsFunc = PsfTraceSizeDiff(shapeCol, psfShapeCol)
+
+    # Holders for multiple visits
+    radiiAll = {k:[] for k in range(1,6) }
+    xipAll = {k:[] for k in range(1,6) }
+    xipErrAll = {k:[] for k in range(1,6) }
+
+    for visit in visits:
+        # print("visit = {}".format(visit))
+        e1, e2 = [], []
+        e1Res, e2Res = [], []
+        SizeRes = []
+        ra = []
+        dec = []
+        notFoundCcds = []
+        for i in range(104):
+            dataId = {"visit": visit, "ccd": i, "filter": filter}
+            try:
+                catalog = butler.get('src', dataId, immediate=True, flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
+            except:
```

Comment:
Can this not work on the combined catalog? Then you don't need a lot of this and can avoid having the butler here. This will make it easier to rebase on top of changes that should hit soon.

Reply:
This was a stand-alone script but it should be able to work with a combined catalog when the rebasing is done.